src/tree.rs

```diff
@@ -621,6 +632,151 @@ async fn compress_file(path: impl AsRef<str>, offset: u64, size: Option<u64>) ->
     }).await
 }
 
+async fn pushlog_handler(
+    rq: RpcMessage,
+    app_state: AppState<State>,
+) -> RpcRequestResult {
+    let log_reader = Log2Reader::new(rq.param().map_or_else(RpcValue::null, RpcValue::clone))
+        .map_err(|e| RpcError::new(RpcErrorCode::InvalidParam, format!("Cannot parse pushLog parameter: {e}")))?;
+    let (since, until) = (log_reader.header.since, log_reader.header.until);
+    let site_path = rq.shv_path().unwrap_or_default();
+    info!("pushLog handler, site: {site_path}, log header: {header}", header = log_reader.header);
+    if !app_state.sites_data.read().await.sites_info.contains_key(site_path) {
+        return Err(RpcError::new(RpcErrorCode::InvalidParam, format!("Wrong site path for pushLog: {site_path}")));
+    }
+
+    let local_journal_path = Path::new(&app_state.config.journal_dir).join(site_path);
+    let journal_file_path = local_journal_path.join(since.to_iso_string() + ".log2");
+
+    let local_latest_entries = {
+        let log_files = get_files(local_journal_path, is_log2_file)
+            .await
+            .map(|mut files| { files.sort_by_key(|entry| std::cmp::Reverse(entry.file_name())); files })
+            .inspect_err(|err| {error!("Cannot read journal dir entries: {err}"); })
+            .unwrap_or_default();
```

Comment:
nit:
```suggestion
        let log_files = get_files(local_journal_path, is_log2_file)
            .await
            .inspect_err(|err| {error!("Cannot read journal dir entries: {err}"); })
            .unwrap_or_default();
        log_files.sort_by_key(|entry| std::cmp::Reverse(entry.file_name()));
```
I know that it's nice to have the sorted list immediately in log_files, but I think using .map like this (mutating the value, and then returning) is a bit weird.

Reply:
I used `map` to delegate the sort to an inner scope and keep `log_file` immutable.